API.md

```diff
@@ -0,0 +1,335 @@
+  squash?: boolean;
+}
+
+export type EditEventV2<E extends Edit = Edit> = CustomEvent<EditDetailV2<E>>;
+
+export type EditEventOptions = {
+  title?: string;
+  squash?: boolean;
+}
+
+export function newEditEventV2<E extends Edit>(edit: E, options: EditEventOptions): EditEventV2<E> {
+  return new CustomEvent<E>('oscd-edit-v2', {
+    composed: true,
+    bubbles: true,
+    detail: {...options, edit},
+  });
+}
+
+declare global {
+  interface ElementEventMap {
+    ['oscd-edit-v2']: EditEventV2;
+  }
+}
+```
+
+Its `title` property is a human-readable description of the edit for displaying in the editing history dialog.
+
+The `squash` flag indicates whether the edit should be merged with the previous edit in the history.
+
+#### `Edit` type
+
+The `EditDetailV2` defined above contains an `edit` of this type:
+
+```typescript
+export type EditV2 = Insert | SetAttributes | SetTextContent | Remove | Edit[];
+```
+
+This means that a single edit can either consist in a sequence of other edits or in one of the following atomic edit types:
+
+> Intent to set or remove (if null) attributes on `element`.
+```typescript
+export type SetAttributes = {
+  element: Element;
+  attributes: Partial<Record<string, string | null>>;
+  attributesNS: Partial<Record<string, Partial<Record<string, string | null>>>>;
+  id?: string; // correlation id
+};
+```
+
+> Intent to set the `textContent` of `element`.
+```typescript
+export type SetTextContent = {
+  element: Element;
+  textContent: string | null;
+  id?: string; // correlation id
+};
+```
+
+> Intent to `parent.insertBefore(node, reference)`
+```typescript
+export type Insert = {
+  parent: Node;
+  node: Node;
+  reference: Node | null;
+  id?: string; // correlation id
+};
+```
+
+> Intent to remove a `node` from its `ownerDocument`.
+```typescript
+export type Remove = {
+  node: Node;
+  id?: string; // correlation id
+};
+```
+
+The optional `id` property is used to correlate the edit with the corresponding undo/redo operation, e.g. in order to check if the edit was successful.
+
+
+### `OpenEvent`
+
+The **open event** allows a plugin to add a document `doc` to the `docs` collection under the name `docName`.
+
+```typescript
+export type OpenDetail = {
+  doc: XMLDocument;
+  docName: string;
+};
+
+export type OpenEvent = CustomEvent<OpenDetail>;
+
+export function newOpenEvent(doc: XMLDocument, docName: string): OpenEvent {
+  return new CustomEvent<OpenDetail>('oscd-open', {
+    bubbles: true,
+    composed: true,
+    detail: { doc, docName },
+  });
+}
+
+declare global {
+  interface ElementEventMap {
+    ['oscd-open']: OpenEvent;
+  }
+}
+```
+
+### `WizardEvent`
+
+The **wizard event** allows the plugin to request opening a modal dialog enabling the user to edit an arbitrary SCL `element`, regardless of how the dialog for editing this particular type of element looks and works.
+
+```typescript
+/* eslint-disable no-undef */
+interface WizardRequestBase {
+  subWizard?: boolean;
```

Comment:
I would be interested to hear other's thoughts on how `subWizard` was intended to be used.
I tried an approach for an `lnode` wizard for mapping `LN`s onto a `LNode` within the substation section of the scd file.

https://github.com/OpenEnergyTools/scl-wizarding/pull/10/files#diff-7b8cd3a9c9d707212e2cb17c3aabdabf7e0e0de291f3b510a01c42b519fac90e

This PR switches to a slightly different version of the `lNode` wizard.

Was this the intended use of this feature?

Reply:
Thanks for your input into this, Jarrad! I'm glad to see you interested in the future of this API.

The intended use is when you want to jump to a different view to let the user edit some detail or dependent piece of data but want to jump back to the currently opened wizard as soon as they have saved or cancelled their changes. This allows us, for example, to jump to editing `DO`s from `LNodeTypes`, and then go back to continuing to edit the same `LNodeType`.

Reply:
Sorry for having you waited so long on the linked review. I will get at it today and will leave comments in the review how `subWizard` works.